app/src/main/java/com/courseproject/tindar/ChatRecyclerViewAdapter.java

```diff
@@ -0,0 +1,145 @@
+package com.courseproject.tindar;
+
+import android.content.Context;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.RelativeLayout;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.courseproject.tindar.entities.MessageModel;
+
+import java.util.ArrayList;
+
+/* TODO: Current implementation of message types (sent vs. received) won't generalize well
+    to future features like file-type messages.
+    In an ideal world, I'd refactor such that each case handles itself,
+    but I'm not currently familiar enough with all the inner workings here
+    and we have other more important features to focus on. Some things to potentially fix:
+    - constants for switch/if-else statements or remove them entirely
+ */
+
+
+/**
+ * I think this feeds into the RecyclerView for the one-on-one chat messages.
+ * Still very confused about what it actually does.
+ * @author Sophia Wan
+ */
+public class ChatRecyclerViewAdapter
+        extends RecyclerView.Adapter<ChatRecyclerViewAdapter.TindarMessageViewHolder>{
+    /**
+     * userID of current user. Needed check whether a message was sent or received.
+     * Might later refactor such that we don't need this.
+     */
+    private final String _userID;
+    /** List of messages to be displayed. Misnomer, but private so it doesn't really matter. */
+    private ArrayList<MessageModel> _displayedMessages;
```

Comment:
can we use an entity in the frontend? I think you need separate model in the use case layer to use it in the frontend.

Reply:
I think it's fine since it's an arraylist of models. If it's not fine then Myles will point it out and we can fix it.